src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -294,15 +294,19 @@ private void OnInventoryReceived(IInventory inventory)
             {
                 case Transaction transaction:
                     system.Consensus?.Tell(transaction);
+                    system.TxRouter.Tell(inventory);
                     break;
                 case Block block:
                     if (block.Index > Blockchain.Singleton.Height + InvPayload.MaxHashesCount) return;
                     UpdateLastBlockIndex(block.Index, false);
+                    system.Blockchain.Tell(inventory);
+                    break;
+                default:
+                    system.Blockchain.Tell(inventory);
                     break;
             }
             knownHashes.Add(inventory.Hash);
```

Comment:
I think that we should move this line to the top

Reply:
have fixed